builder/azure/arm/builder.go

```diff
@@ -206,12 +206,22 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 		if err != nil {
 			return nil, fmt.Errorf("the Shared Gallery Image '%s' to which to publish the managed image version to does not exist in the resource group '%s' or does not contain managed image '%s'", b.config.SharedGalleryDestination.SigDestinationGalleryName, b.config.SharedGalleryDestination.SigDestinationResourceGroup, b.config.SharedGalleryDestination.SigDestinationImageName)
 		}
-
 		// Check if a Image Version already exists for our target destination
 		galleryImageVersionId := galleryimageversions.NewImageVersionID(sigSubscriptionID, b.config.SharedGalleryDestination.SigDestinationResourceGroup, b.config.SharedGalleryDestination.SigDestinationGalleryName, b.config.SharedGalleryDestination.SigDestinationImageName, b.config.SharedGalleryDestination.SigDestinationImageVersion)
 		_, err := azureClient.GalleryImageVersionsClient.Get(ctx, galleryImageVersionId, galleryimageversions.DefaultGetOperationOptions())
 		if err == nil {
-			return nil, fmt.Errorf("a gallery image version for image name:version %s:%s already exists in gallery %s", b.config.SharedGalleryDestination.SigDestinationImageName, b.config.SharedGalleryDestination.SigDestinationImageVersion, b.config.SharedGalleryDestination.SigDestinationGalleryName)
+			if b.config.PackerForce {
+				ui.Say(fmt.Sprintf("a gallery image version for image name:version %s:%s already exists in gallery %s, but deleting it due to -force flag", b.config.SharedGalleryDestination.SigDestinationGalleryName, b.config.SharedGalleryDestination.SigDestinationImageVersion, b.config.SharedGalleryDestination.SigDestinationImageName))
+				deleteImageContext, cancel := context.WithTimeout(ctx, azureClient.PollingDuration)
+				defer cancel()
```

Comment:
Why the `defer cancel`? I'm not sure I understand the rationale behind this one, if we quit the `Run` function at that point the context is either expired because it was already cancelled, or the build has finished and it will be ignored then since there's nothing left to cancel. Unless the function does something else and I'm missing something?

Reply:
It's just communicating to the SDK to stop accepting new requests from that context 🤷 I think its just best practice to cancel TimeOut contexts you create

Reply:
Could be best practice yes, no objection to do it then, but if that's systematic for the `Run` function, I'd think it can be done once when the function starts executing, and we don't need to repeat that deferred call around the various paths of the code?

Reply:
We talked about this privately but just to post here for anyones context reading in the future, but there was some confusion, the cancel function isn't global and is returned by the With Context function, closing this since we talked about it